src/adapter/aws-lambda/handler.ts

```diff
@@ -198,16 +200,25 @@ const createRequest = (event: LambdaEvent) => {
   const domainName =
     event.requestContext && 'domainName' in event.requestContext
       ? event.requestContext.domainName
-      : event.headers['host']
+      : event.headers?.['host'] ?? event.multiValueHeaders?.['host']?.[0]
   const path = isProxyEventV2(event) ? event.rawPath : event.path
   const urlPath = `https://${domainName}${path}`
   const url = queryString ? `${urlPath}?${queryString}` : urlPath
 
   const headers = new Headers()
   getCookies(event, headers)
-  for (const [k, v] of Object.entries(event.headers)) {
-    if (v) {
-      headers.set(k, v)
+  if (event.headers) {
+    for (const [k, v] of Object.entries(event.headers)) {
+      if (v) {
+        headers.set(k, v)
+      }
+    }
+  }
+  if (event.multiValueHeaders) {
```

Comment:
Note) This is not `} else if ` because `headers` are not optional, at least in type definition.